app/services/ai_backend/gemini.rb

```diff
@@ -0,0 +1,116 @@
+class AIBackend::Gemini < AIBackend
+  include Tools
+
+  # Rails system tests don't seem to allow mocking because the server and the
+  # test are in separate processes.
+  #
+  # In regular tests, mock this method or the TestClients::Gemini class to do
+  # what you want instead.
+  def self.client
+    if Rails.env.test?
+      ::TestClient::Gemini
+    else
+      ::Gemini
+    end
+  end
+
+  def initialize(user, assistant, conversation = nil, message = nil)
+    super(user, assistant, conversation, message)
+    begin
+      raise ::OpenAI::ConfigurationError if assistant.api_service.requires_token? && assistant.api_service.effective_token.blank?
+      Rails.logger.info "Connecting to Gemini API server at #{assistant.api_service.url} with access token of length #{assistant.api_service.effective_token.to_s.length}"
+      @client = self.class.client.new(credentials: {service: "generative-language-api",
+                                                    api_key: assistant.api_service.effective_token,
+                                                    version: "v1beta"},
+                                      options: { model: assistant.language_model.api_name,
+                                      server_sent_events: true })
+    rescue ::Faraday::UnauthorizedError => e
+      raise OpenAI::ConfigurationError
+    end
+  end
+
+  def client_method_name
+    :stream_generate_content
+  end
+
+  def configuration_error
+    ::OpenAI::ConfigurationError
```

Comment:
```suggestion
    class ::Gemini::Errors::ConfigurationError
```

And inside get_next_ai_message_job catch this error and display a pretty output, you'll see the pattern with the others.

Try removing your Gemini API key from your HostedGPT settings and ensure it displays a pretty message. Also, try intentionally putting a bad Gemini api key into your HostedGPT settings and see if it throws a different exception. You likely need to catch that one too inside get_next_ai_job.